app/models/concerns/sortable.rb

```diff
@@ -0,0 +1,75 @@
+# The code in here is a bit complicated.
+# I wanted to be able to use include with arguments.
+# Read this blogpost: https://cutt.ly/jUr8M8f
+class Sortable < Module
+
+  #
+  # *fields are the fields which can be sorted
+  # 
+  def initialize(*fields)
+    # super() with brackets calls super with no arguments.
+    # Otherwise the would get passed and would throw an error
+    super() do
+
+      # Create an anonymous Module and store it in a local variable
+      # It will contain all the class methods which are later appended
+      # to the ActiveRecord Model.
+      #
+      # For example:
+      # class Model < ActiveRecord::Base
+      #   include Sortable.new :name, :email
+      # end
+      #
+      # Would lead to:
+      # Model.sorted_by_name(order)
+      # Model.sorted_by_email(order)
+      singleton_methods = Module.new do
+        fields.each do |field|
+          method_name = "sorted_by_#{field}".to_sym
+          define_method(method_name) do |order|
+            p = Hash.new
```

Comment:
bitte keine Variablendefinitionen mit nur einem Buchstaben, damit man es auch in 6 Monaten noch schnell versteht :-)

Reply:
done